flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFATest.java

```diff
@@ -324,6 +328,89 @@ public boolean filter(Event value) throws Exception {
 		}
 	}
 
+	@Test
+	public void testNFAChange() {
+		Pattern<Event, ?> pattern = Pattern.<Event>begin("start").where(new SimpleCondition<Event>() {
+			private static final long serialVersionUID = 1858562682635302605L;
```

Comment:
Anyway I feel it would be easier to read and understand if we extract a test for a `nfaChange` case at a time.